napari/_qt/dialogs/_tests/test_qt_plugin_dialog.py

```diff
@@ -1,14 +1,19 @@
+import importlib.metadata
 from typing import Generator, Optional, Tuple
+from unittest.mock import patch
 
+import npe2
 import pytest
-from npe2 import PackageMetadata
 
+import napari.plugins
 from napari._qt.dialogs import qt_plugin_dialog
+from napari._qt.dialogs.qt_package_installer import InstallerActions
+from napari.plugins._tests.test_npe2 import mock_pm  # noqa
 
 
 def _iter_napari_pypi_plugin_info(
     conda_forge: bool = True,
-) -> Generator[Tuple[Optional[PackageMetadata], bool], None, None]:
+) -> Generator[Tuple[Optional[npe2.PackageMetadata], bool], None, None]:
```

Comment:
As this whole function is used in thread and codecov has a problem with the collection of coverage in such cases, maybe this should be excluded from test coverage (`# pargma: no cover` plus information about the problem with coverage when using threading)